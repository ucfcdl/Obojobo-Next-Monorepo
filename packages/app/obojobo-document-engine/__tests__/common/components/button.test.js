import React from 'react'
import Button from '../../../src/scripts/common/components/button'
import renderer from 'react-test-renderer'

// jest.mock('../../../src/scripts/common/util/isornot')
jest.mock('../../../src/scripts/common/page/focus')

describe('Button', () => {
	test('Renders', () => {
		const component = renderer.create(<Button className="testClassName">Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('dangerous', () => {
		const component = renderer.create(<Button isDangerous>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('altAction', () => {
		const component = renderer.create(<Button altAction>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('align', () => {
		const component = renderer.create(<Button align="left">Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('shouldPreventTab', () => {
		const component = renderer.create(<Button shouldPreventTab={true}>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('disabled', () => {
		const component = renderer.create(<Button disabled={true}>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('isSubmittable', () => {
		const component = renderer.create(<Button isSubmittable={true}>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('isSubmittable', () => {
		const component = renderer.create(<Button isSubmittable={true}>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('with value', () => {
		const component = renderer.create(<Button value="child value">Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('with null value', () => {
		const component = renderer.create(<Button value={null}>Label</Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('with empty string value', () => {
		const component = renderer.create(<Button value={''}></Button>)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('focus', () => {
		const focus = require('../../../src/scripts/common/page/focus').default
		const component = renderer.create(<Button value={''}></Button>)
		expect(focus).not.toHaveBeenCalled()
		component.getInstance().focus()
		expect(focus).toHaveBeenCalled()
	})
})
