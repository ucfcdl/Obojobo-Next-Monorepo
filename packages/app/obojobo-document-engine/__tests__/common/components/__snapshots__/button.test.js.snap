// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<div
  className="obojobo-draft--components--button is-not-dangerous align-center testClassName"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
  >
    Label
  </button>
</div>
`;

exports[`Button align 1`] = `
<div
  className="obojobo-draft--components--button is-not-dangerous align-left"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
  >
    Label
  </button>
</div>
`;

exports[`Button altAction 1`] = `
<div
  className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
  >
    Label
  </button>
</div>
`;

exports[`Button dangerous 1`] = `
<div
  className="obojobo-draft--components--button is-dangerous align-center"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
  >
    Label
  </button>
</div>
`;

exports[`Button disabled 1`] = `
<div
  className="obojobo-draft--components--button is-not-dangerous align-center"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
    disabled={true}
  >
    Label
  </button>
</div>
`;

exports[`Button shouldPreventTab 1`] = `
<div
  className="obojobo-draft--components--button is-not-dangerous align-center"
  contentEditable={false}
>
  <button
    className="button"
    contentEditable={false}
    disabled={true}
    tabIndex="-1"
  >
    Label
  </button>
</div>
`;
